roles/capi_cluster/tasks/main.yml

```diff
@@ -102,8 +102,95 @@
         content: "{{ capi_cluster_kubeconfig_cmd.stdout | b64decode }}"
         dest: "{{ capi_cluster_kubeconfig_path }}"
         mode: u=rw,g=,o=
-  when: capi_cluster_release_state == 'present'
-
+      
+    - name: Create namespace for a new service account
+      kubernetes.core.k8s:
+        api_version: v1
+        kind: Namespace
+        name: kubeconfig-sa-namespace
+        state: present
+      delay: 10
+      
+    - name: Create service account for generating kubeconfig
+      kubernetes.core.k8s:
+        api_version: v1
+        kind: ServiceAccount
+        namespace: kubeconfig-sa-namespace
+        name: kubeconfig-service-account
+        state: present
+      delay: 10
+      
+    - name: Create secret for the service account
+      kubernetes.core.k8s:
+        state: present
+        definition:
+          apiVersion: v1
+          kind: Secret
+          type: kubernetes.io/service-account-token
+          metadata:
+            name: kubeconfig-secret
+            namespace: kubeconfig-sa-namespace
+            annotations:
+              kubernetes.io/service-account.name: kubeconfig-service-account
+      delay: 10
+      
+    - name: Create clusterrole binding of cluster-admin configuration for kubeconfig service account
+      kubernetes.core.k8s:
+        state: present
+        definition:
+          apiVersion: rbac.authorization.k8s.io/v1
+          kind: ClusterRoleBinding
+          metadata:  
+            name: kubeconfig-role-binding
+            namespace: kubeconfig-sa-namespace
+          subjects:
+            - kind: ServiceAccount
+              name: kubeconfig-service-account
+              namespace: kubeconfig-sa-namespace
+          roleRef:
+            kind: ClusterRole
+            name: cluster-admin
+            apiGroup: rbac.authorization.k8s.io
```

Comment:
Can we add a comment here explaining why we need to bind to the cluster-admin role please and also explain why making a separate restricted role is not worth it (since the role would need permission to create new roles and service accounts anyway).